### PR TITLE
Add support for generating scaladoc documentation using doc command

### LINKED ISSUE #942: Add doc command to generate documentation via underlying scaladoc
**Is your feature request related to a problem? Please describe.**
It would be really nice to be able to generate documentation without having to switch to sbt or install the scala binaries manually.

**Describe the solution you'd like**
Let `scala-cli doc .` generate documentation for the code in the current dir, for example written to a dir named `doc` or similar.

**Additional context**
This would mean that scala-cli needs to install and call scaladoc. There are also various options to scaladoc that somehow should be possible to control from `scala-cli doc`

### PR DESCRIPTION
**Pull request by lwronski**

### CODE REVIEW

**Comment by alexarchambault** [comment]
I'm not sure this should be an option on the `package` command. We don't really package things here, from a user's perspective (even if we were to rely on javadoc JAR caching internally, it's only an implementation detail, users don't need to be aware of it).

To me, this would be better either as an option on `compile` (like `scala-cli compile . --doc --output dest/`), or as a separate `doc` command.

**Comment by bjornregnell** [comment]
I think a separate doc command is the most natural and easiest to remember and use.
`scala-cli doc .`

File: website/docs/commands/doc.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+title: Doc
+sidebar_position: 17
+---
+
+Scala CLI can generate the API documentation of your Scala 2 and 3 projects. It provides similar features to `javadoc`.
```
Comment (alexarchambault) [inline website/docs/commands/doc.md]: ?
```suggestion
Scala CLI can generate the API documentation of your Scala 2, Scala 3, and Java projects. It provides features similar to `javadoc`.
```

File: website/docs/commands/doc.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ Scaladoc documentation is generated in directory with static site files:
```
Comment (alexarchambault) [inline website/docs/commands/doc.md]:
```suggestion
The API documentation is generated in a directory whose files make up a static website:
```

File: website/docs/commands/doc.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+```bash 
+scala-cli doc Hello.scala -o scala-doc 
+# Wrote Scaladoc to ./scala-doc
+```
+
+<!-- Expected
+Wrote Scaladoc to ./scala-doc
+-->
+
+Output directory `scala-doc` contains static site files with your documentation.
```
Comment (alexarchambault) [inline website/docs/commands/doc.md]:
```suggestion
The output directory `scala-doc` contains the static site files with your documentation.
```

File: website/docs/commands/doc.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+After opening generated static documentation (you have to open `scala-doc/index.html` in your browser),
```
Comment (alexarchambault) [inline website/docs/commands/doc.md]:
```suggestion
After opening the generated static documentation (you have to open `scala-doc/index.html` in your browser), 
```

File: website/docs/commands/doc.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+<!-- Expected
+Wrote Scaladoc to ./scala-doc
+-->
+
+Output directory `scala-doc` contains static site files with your documentation.
+
+After opening generated static documentation (you have to open `scala-doc/index.html` in your browser), 
+you will see generated Scaladoc documentation. The following screen shows the definition `main` method:
```
Comment (alexarchambault) [inline website/docs/commands/doc.md]:
```suggestion
you will see the generated scaladoc documentation. The following screen shows the definition of the `main` method:
```

**Review by alexarchambault — COMMENTED** [review]
Looks great! I only left minor comments about phrasing in the added documentation.

**Review by alexarchambault — APPROVED** [review]

**Comment by bjornregnell** [comment]
:tada:

**Pull request merged**